Sources/Starknet/Data/StarknetChainId.swift

```diff
@@ -1,16 +1,20 @@
 import BigInt
 import Foundation
 
-public enum StarknetChainId {
-    case mainnet
-    case testnet
+public enum StarknetChainId: String, Codable, Equatable {
+    case mainnet = "0x534e5f4d41494e"
+    case goerli = "0x534e5f474f45524c49"
+    case sepolia_testnet = "0x534e5f5345504f4c4941"
+    case sepolia_integration = "0x534e5f494e544547524154494f4e5f5345504f4c4941"
 
     public var feltValue: Felt {
-        switch self {
-        case .mainnet:
-            return Felt(fromHex: "0x534e5f4d41494e")!
-        case .testnet:
-            return Felt(fromHex: "0x534e5f474f45524c49")!
-        }
+        Felt(fromHex: self.rawValue)!
+    }
+
+    enum CodingKeys: String, CodingKey {
+        case mainnet = "0x534e5f4d41494e"
+        case goerli = "0x534e5f474f45524c49"
+        case sepolia_testnet = "0x534e5f5345504f4c4941"
+        case sepolia_integration = "0x534e5f494e544547524154494f4e5f5345504f4c4941"
     }
```

Comment:
Not actually sure about having `StarknetChainId` enum, now that we don't have to pass it to the `StarknetProvider` constructors? 
 
It's more explicit if someone checks chainId, but I'm not sure there are users that don't know what kind of node they're using 😄 
Might as well use Felt instead, and avoid the need to make changes in an unlikely event someone needs to use custom chain ids.  

WDYT?

Reply:
Do we need it elsewhere? for coding/decoding or for some verification? I suppose chainId endpoint returns hexes, so if we try to communicate about these ids to users somewhere, we might want to use user-readable names

Reply:
I think we only use those to for signing transactions

Reply:
If that's the case we could probably remove it

Reply:
Will be addressed in #138